numpy/ma/core.py

```diff
@@ -1224,6 +1227,177 @@ def __str__(self):
 fmod = _DomainedBinaryOperation(umath.fmod, _DomainSafeDivide(), 0, 1)
 mod = _DomainedBinaryOperation(umath.mod, _DomainSafeDivide(), 0, 1)
 
+# Include masked dot here as it is part of the __numpy_ufunc__ system. This
```

Comment:
`dot`, both method and function, as well as `matmul` (@) check `__numpy_ufunc__`, although none of them are ufuncs ;) I don't have a problem with moving the definitions, I wasn't sure where the best spot was.